WcaOnRails/app/webpacker/components/CompetitionOverview/CompetitionFilters.js

```diff
@@ -0,0 +1,662 @@
+  const [mapDisplayComps, setMapDisplayComps] = useState([]);
+
+  const editPastComps = (comps, year) => {
+    setPastComps((prevPastComps) => ({
+      ...prevPastComps,
+      [year]: comps,
+    }));
+  };
+
+  const {
+    data: competitionsData,
+    fetchNextPage: competitionsFetchNextPage,
+    isFetching: competitionsIsFetching,
+    hasNextPage: hasMoreCompsToLoad,
+  } = useInfiniteQuery({
+    queryKey: ['competitions', competitionApiKey],
+    queryFn: ({ pageParam = 1 }) => {
+      const dateNow = new Date(Date.now());
+      let searchParams;
+
+      if (timeOrder === 'present') {
+        searchParams = new URLSearchParams({
+          sort: 'start_date,end_date,name',
+          ongoing_and_future: dateNow.toISOString().split('T')[0],
+          page: pageParam,
+        });
+      } else if (timeOrder === 'recent') {
+        const thirtyDaysAgo = new Date();
+        thirtyDaysAgo.setDate(dateNow.getDate() - 30);
+
+        searchParams = new URLSearchParams({
+          sort: '-end_date,-start_date,name',
+          start: thirtyDaysAgo.toISOString().split('T')[0],
+          end: dateNow.toISOString().split('T')[0],
+          page: pageParam,
+        });
+      } else if (timeOrder === 'past') {
+        if (pastSelectedYear === 'all_years') {
+          searchParams = new URLSearchParams({
+            sort: '-end_date,-start_date,name',
+            end: dateNow.toISOString().split('T')[0],
+            page: pageParam,
+          });
+        } else {
+          searchParams = new URLSearchParams({
+            sort: '-end_date,-start_date,name',
+            start: `${pastSelectedYear}-1-1`,
+            end: dateNow.getFullYear() === pastSelectedYear ? dateNow.toISOString().split('T')[0] : `${pastSelectedYear}-12-31`,
+            page: pageParam,
+          });
+        }
+      } else if (timeOrder === 'by_announcement') {
+        searchParams = new URLSearchParams({
+          sort: '-announced_at,name',
+          page: pageParam,
+        });
+      } else if (timeOrder === 'custom') {
+        searchParams = new URLSearchParams({
+          sort: 'start_date,end_date,name',
+          start: customStartDate?.toISOString().split('T')[0] || '',
+          end: customEndDate?.toISOString().split('T')[0] || '',
+          page: pageParam,
+        });
+      }
+
+      if (selectedRegion) {
+        // Continent IDs begin with underscore
+        const regionParam = selectedRegion[0] === '_' ? 'continent' : 'country_iso2';
+        searchParams.append(regionParam, selectedRegion);
+      }
+      if (selectedDelegate) {
+        searchParams.append('delegate', selectedDelegate);
+      }
+      if (searchQuery) {
+        searchParams.append('q', searchQuery);
+      }
+
+      return fetchJsonOrError(`${competitionsApiUrl}?${searchParams.toString()}`);
+    },
+    getNextPageParam: (lastPage, allPages) => {
+      // Assuming the last page has less than the max number of competitions fetched per query
```

Comment:
The latter

Reply:
So the `lastPage` param here is the most recent page fetched, not the actual last page of data? I see that the infinite query documentation uses `lastPage`, but `previousPage` seems more logical to me, especially since our infinite scroll always has a last page.

I don't know if it's just me, but if a comment says that it's assuming something, I assume the code may potentially break or behave unexpectedly if the assumption fails. There's no reason that the back-end would send less than a full page unless it's the end of the results right?

Reply:
I'd say it's clearer to frame this as "if we get a full page of results back, then there may be another page" rather than "if we don't get a full page back then we assume there are no more pages".

Minor difference, but it might help whoever is modifying this code a few years from now.

Reply:
Sounds good, I see where you're coming from